File: sample/src/main/java/com/u1f4f1/sample/SampleAdapter.java
package com.u1f4f1.sample;

import com.u1f4f1.powerbottomsheet.bottomsheet.BottomSheetAdapter;
import com.u1f4f1.powerbottomsheet.coordinatorlayoutbehaviors.AnchorPointBottomSheetBehavior;

import static com.u1f4f1.powerbottomsheet.LogKt.trace;

public class SampleAdapter extends BottomSheetAdapter {
    public SampleAdapter(AnchorPointBottomSheetBehavior<?> behavior) {
        super(behavior);
    }

    public void addFakeViews() {
        trace("adding fake views to adapter");
        for (int i = 0; i < 10; i++) {
            addModel(new HoursModel_().buttonClickConsumer(recyclerViewTransitionRunnable));
        }
    }
}

File: sample/src/main/java/com/u1f4f1/sample/SampleAdapter.kt
package com.u1f4f1.sample

import com.mooveit.library.Fakeit
import com.u1f4f1.powerbottomsheet.bottomsheet.BottomSheetAdapter
import com.u1f4f1.powerbottomsheet.coordinatorlayoutbehaviors.AnchorPointBottomSheetBehavior

class SampleAdapter(behavior: AnchorPointBottomSheetBehavior<*>?) : BottomSheetAdapter(behavior) {

    val headerModel = HeaderModel_()

    fun addHeader() = addModel(headerModel)

    fun addFakeViews() {
        addModel(ExpandingCardModel_().delayedTransitionRunnable { recyclerViewTransitionRunnable.run() })

        for (i in 0..10) {
            addModel(QuoteCardModel_().quote(Fakeit.rickAndMorty().quote()).attribution(Fakeit.rickAndMorty().character()))
        }
    }

    fun activateHeader(activate: Boolean) {
        headerModel.active = activate
        notifyModelChanged(headerModel)
    }
}
